components/sup/src/manager/mod.rs

```diff
@@ -696,10 +705,20 @@ impl Manager {
                 self.shutdown(ShutdownReason::Departed);
                 return Err(sup_error!(Error::Departed));
             }
-            if let Some(SignalEvent::Passthrough(Signal::HUP)) = signals::check_for_signal() {
-                outputln!("Supervisor shutting down for signal");
-                self.shutdown(ShutdownReason::Signal);
-                return Ok(());
+            if !feat::is_enabled(feat::IgnoreSignals) {
+                match signals::check_for_signal() {
+                    Some(SignalEvent::Passthrough(Signal::HUP)) => {
+                        outputln!("Supervisor shutting down for signal");
+                        self.shutdown(ShutdownReason::Signal);
+                        return Ok(());
+                    }
+                    Some(SignalEvent::Shutdown) => {
```

Comment:
Why are we making this change? The supervisor isn't supposed to receive these signals. Shutdown needs to be orchestrated through the launcher and [the launcher communicates that via the IPC channel here](https://github.com/habitat-sh/habitat/blob/master/components/launcher/src/server/mod.rs#L180).

Arguably the catching of the `HUP` signal here should've been done in the launcher and passed to the supervisor through IPC, but it was less code to just let the signal pass through directly. Adding additional signal handling the Supervisor is something worth discussing, but I don't think it should be part of this change.